indico/web/client/js/react/components/WTFDateTimeField.jsx

```diff
@@ -0,0 +1,237 @@
+
+  useEffect(() => {
+    if (!timezoneField) {
+      return;
+    }
+    function handleTimezone() {
+      setTimezone(timezoneField.value);
+    }
+    timezoneField.addEventListener('change', handleTimezone);
+
+    return () => {
+      timezoneField.removeEventListener('change', handleTimezone);
+    };
+  }, [timezoneField]);
+
+  useEffect(() => {
+    timePickerRef.current.picker.required = required;
+  }, [required]);
+
+  useEffect(() => {
+    if (!linkedField) {
+      return;
+    }
+    function handleDateChange() {
+      const linkedDate = moment(linkedFieldDateElem.value, 'DD/MM/YYYY');
+      if (
+        (linkedField.notBefore && linkedDate.isAfter(date, 'day')) ||
+        (linkedField.notAfter && linkedDate.isBefore(date, 'day'))
+      ) {
+        updateDate(linkedDate);
+      }
+    }
+    linkedFieldDateElem.addEventListener('change', handleDateChange);
+
+    return () => {
+      linkedFieldDateElem.removeEventListener('change', handleDateChange);
+    };
+  }, [date, linkedFieldDateElem, linkedField, updateDate]);
+
+  const clearFields = () => {
+    updateTime(null);
+    updateDate(null);
+    clearRef.current.dispatchEvent(new Event('indico:closeAutoTooltip'));
+  };
+
+  const getDisabledHours = useCallback(() => {
+    const hours = [];
+    if (earliestMoment && date.isSame(earliestMoment, 'day')) {
+      hours.push([...Array(earliestMoment.hour()).keys()]);
+    }
+    if (latestMoment && date.isSame(latestMoment, 'day')) {
+      const hourInt = latestMoment.hour();
+      hours.push([...Array(23 - hourInt).keys()].map(x => x + hourInt + 1));
+    }
+    return hours.flat();
+  }, [earliestMoment, latestMoment, date]);
+
+  const getDisabledMinutes = useCallback(
+    h => {
+      const minutes = [];
+      if (earliestMoment && date.isSame(earliestMoment, 'day') && h === earliestMoment.hour()) {
+        minutes.push([...Array(earliestMoment.minutes()).keys()]);
+      }
+      if (latestMoment && date.isSame(latestMoment, 'day') && h === latestMoment.hour()) {
+        const minInt = latestMoment.minutes();
+        minutes.push([...Array(59 - minInt).keys()].map(x => x + minInt + 1));
+      }
+      return minutes.flat();
+    },
+    [earliestMoment, latestMoment, date]
+  );
+
+  const isOutsideRange = useCallback(
+    value => {
+      if (linkedField) {
+        const linkedDate = moment(linkedFieldDateElem.value, 'DD/MM/YYYY');
+        if (linkedField.notBefore && value.isBefore(linkedDate, 'day')) {
+          return true;
+        }
+        if (linkedField.notAfter && value.isAfter(linkedDate, 'day')) {
+          return true;
+        }
+      }
+      if (earliestMoment && value.isBefore(earliestMoment, 'day')) {
+        return true;
+      }
+      if (latestMoment && value.isAfter(latestMoment, 'day')) {
+        return true;
+      }
+      return false;
+    },
+    [earliestMoment, latestMoment, linkedField, linkedFieldDateElem]
+  );
+
+  return (
+    <>
+      <SingleDatePicker
```

Comment:
This is probably still under test, but worth mentioning that the date picker widget overflow is hidden. We need a way to allow it to overflow the modal.
![image](https://user-images.githubusercontent.com/12183954/80230533-f5a22400-8649-11ea-9a94-a3592a22ac5c.png)

Reply:
Yes, that was what I still needed to do and why it was a draft :)